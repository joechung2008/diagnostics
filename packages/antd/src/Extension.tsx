import { Typography } from "antd";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Typography.Title level={1}>{extensionName}</Typography.Title>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
