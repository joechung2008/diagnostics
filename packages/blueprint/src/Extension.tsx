import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div id="extension" className="extension-root grow">
      <h2 className="bp6-heading">{extensionName}</h2>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
