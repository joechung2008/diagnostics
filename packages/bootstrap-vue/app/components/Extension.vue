<template>
  <div class="extension">
    <h1>{{ extensionName }}</h1>
    <Configuration v-if="config" :config="config" />
    <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
  </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types/Extension";

defineProps<ExtensionProps>();
</script>

<style scoped>
.extension {
  display: flex;
  flex: 1;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
  padding: 0.25rem;
}
</style>
