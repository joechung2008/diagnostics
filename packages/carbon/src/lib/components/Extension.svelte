<script lang="ts">
	import Configuration from './Configuration.svelte';
	import StageDefinition from './StageDefinition.svelte';

	let { extensionName, config, stageDefinition }: ExtensionProps = $props();
</script>

<div class="extension-root" role="tabpanel" tabindex="0">
	<h1>{extensionName}</h1>
	{#if config}
		<Configuration {config} />
	{/if}
	{#if stageDefinition}
		<StageDefinition {stageDefinition} />
	{/if}
</div>

<style>
	.extension-root {
		display: flex;
		flex: 1;
		flex-direction: column;
		gap: 0.5rem;
		overflow-y: auto;
		padding: 0.5rem;
	}
</style>
