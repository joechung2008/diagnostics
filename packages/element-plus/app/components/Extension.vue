<template>
    <div class="extension">
        <h1>{{ extensionName }}</h1>
        <Configuration v-if="config" :config="config" />
        <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
    </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types"

defineProps<ExtensionProps>()
</script>

<style scoped>
.extension {
    display: flex;
    flex: 1;
    flex-direction: column;
    gap: 0.5rem;
    max-height: max-content;
    overflow-y: auto;
    padding: 0.25rem;
}

h1 {
    margin: 0;
}
</style>
