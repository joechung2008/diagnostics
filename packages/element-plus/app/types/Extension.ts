import type { ExtensionInfo } from "./ExtensionInfo"
import type { ExtensionError } from "./ExtensionError"

export type Extension = ExtensionInfo | ExtensionError
