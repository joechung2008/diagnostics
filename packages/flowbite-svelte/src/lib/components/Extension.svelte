<script lang="ts">
  import { Heading } from "flowbite-svelte";
  import Configuration from "./Configuration.svelte";
  import StageDefinition from "./StageDefinition.svelte";

  let { config, extensionName, stageDefinition }: ExtensionProps = $props();
</script>

<div class="flex flex-grow flex-col gap-0.5 overflow-y-auto p-1">
  <Heading tag="h1">
    {extensionName}
  </Heading>
  {#if config}
    <Configuration {config} />
  {/if}
  {#if stageDefinition}
    <StageDefinition {stageDefinition} />
  {/if}
</div>
