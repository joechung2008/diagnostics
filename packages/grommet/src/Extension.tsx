import { Heading } from "grommet";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Heading level={2} margin="none">
        {extensionName}
      </Heading>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
