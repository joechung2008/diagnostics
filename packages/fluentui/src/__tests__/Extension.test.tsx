import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Extension from "../Extension";

const mockProps = {
  extensionName: "Test Extension",
  config: { foo: "bar" },
  stageDefinition: { build: ["compile", "test"] },
};

describe("Extension", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Extension {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
