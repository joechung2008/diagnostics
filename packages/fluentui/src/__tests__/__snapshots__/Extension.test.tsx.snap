// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="extension-root grow"
  >
    <span
      class="fui-Text ___7xffiy0_wrqr9g0 fk6fouc f1x0m3f5 fb86gi6 fl43uef fpgzoln ftgm304 f6juhto f1gl81tg f2jf649"
    >
      Test Extension
    </span>
    <div>
      <span
        class="fui-Text ___i24qii0_1d5rd8v fk6fouc fod5ikn faaz57k fl43uef fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
      >
        Configuration
      </span>
      <table
        aria-label="Configuration"
        class="fui-Table ___422ui00_e7lxxb0 fgkb47j fhovq9v f1w4nmp0 fmrv4ls fly5x3f f14m3nip"
      >
        <thead
          class="fui-TableHeader ___1gw3ddd_19kkhse f1tp1avn"
        >
          <tr
            class="fui-TableRow ___ktc86j0_5zdzqp0 f19n0e5 f1ewtqcl f1jazu75 f1xeqee6 f1dxfoyt f2krc9w f1facbz3 f1u0rzck"
          >
            <th
              class="fui-TableHeaderCell ___zddznz0_a5pc7o0 figsok6 f3gpkru f14ym4q2 f1euou18 f10pi13n f15pt5es fmrv4ls"
            >
              <div
                class="fui-TableHeaderCell__button ___1g9gf5i_12fejcs fq6nmtn f1e4lqlz f1u2r49w f1ym3bx4 f1mo0ibp fjoy568 fytdu2e f1gl81tg f1mk8lai f3bhgqh fgusgyc f10pi13n fly5x3f f22iagw f1l02sjl f122n59 fkln5zr f1nxs5xn f1izfyrr f1s6fcnf"
                role="presentation"
              >
                Key
              </div>
            </th>
            <th
              class="fui-TableHeaderCell ___zddznz0_a5pc7o0 figsok6 f3gpkru f14ym4q2 f1euou18 f10pi13n f15pt5es fmrv4ls"
            >
              <div
                class="fui-TableHeaderCell__button ___1g9gf5i_12fejcs fq6nmtn f1e4lqlz f1u2r49w f1ym3bx4 f1mo0ibp fjoy568 fytdu2e f1gl81tg f1mk8lai f3bhgqh fgusgyc f10pi13n fly5x3f f22iagw f1l02sjl f122n59 fkln5zr f1nxs5xn f1izfyrr f1s6fcnf"
                role="presentation"
              >
                Value
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="fui-TableBody ___1gw3ddd_19kkhse f1tp1avn"
        >
          <tr
            class="fui-TableRow ___1195ssy_2eyo590 f19n0e5 f1ewtqcl f1jazu75 f1xeqee6 f1dxfoyt f2krc9w f1wfn5kd f1g4hkjv f15ngxrw fjbbrdp f1t94bn6 feu1g3u f1uorfem fw60kww f4xjyn1 ff1wgvm fiob0tu f1j6scgf f1x4h75k f1facbz3 f1u0rzck"
          >
            <td
              class="fui-TableCell ___mwjzo80_69n7eg0 f10pi13n f3gpkru f1dxfoyt f2krc9w f15pt5es fmrv4ls f1ft4266"
            >
              foo
            </td>
            <td
              class="fui-TableCell ___mwjzo80_69n7eg0 f10pi13n f3gpkru f1dxfoyt f2krc9w f15pt5es fmrv4ls f1ft4266"
            >
              bar
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div>
      <span
        class="fui-Text ___i24qii0_1d5rd8v fk6fouc fod5ikn faaz57k fl43uef fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
      >
        Stage Definitions
      </span>
      <table
        aria-label="Stage Definitions"
        class="fui-Table ___422ui00_e7lxxb0 fgkb47j fhovq9v f1w4nmp0 fmrv4ls fly5x3f f14m3nip"
      >
        <thead
          class="fui-TableHeader ___1gw3ddd_19kkhse f1tp1avn"
        >
          <tr
            class="fui-TableRow ___ktc86j0_5zdzqp0 f19n0e5 f1ewtqcl f1jazu75 f1xeqee6 f1dxfoyt f2krc9w f1facbz3 f1u0rzck"
          >
            <th
              class="fui-TableHeaderCell ___zddznz0_a5pc7o0 figsok6 f3gpkru f14ym4q2 f1euou18 f10pi13n f15pt5es fmrv4ls"
            >
              <div
                class="fui-TableHeaderCell__button ___1g9gf5i_12fejcs fq6nmtn f1e4lqlz f1u2r49w f1ym3bx4 f1mo0ibp fjoy568 fytdu2e f1gl81tg f1mk8lai f3bhgqh fgusgyc f10pi13n fly5x3f f22iagw f1l02sjl f122n59 fkln5zr f1nxs5xn f1izfyrr f1s6fcnf"
                role="presentation"
              >
                Key
              </div>
            </th>
            <th
              class="fui-TableHeaderCell ___zddznz0_a5pc7o0 figsok6 f3gpkru f14ym4q2 f1euou18 f10pi13n f15pt5es fmrv4ls"
            >
              <div
                class="fui-TableHeaderCell__button ___1g9gf5i_12fejcs fq6nmtn f1e4lqlz f1u2r49w f1ym3bx4 f1mo0ibp fjoy568 fytdu2e f1gl81tg f1mk8lai f3bhgqh fgusgyc f10pi13n fly5x3f f22iagw f1l02sjl f122n59 fkln5zr f1nxs5xn f1izfyrr f1s6fcnf"
                role="presentation"
              >
                Value
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="fui-TableBody ___1gw3ddd_19kkhse f1tp1avn"
        >
          <tr
            class="fui-TableRow ___1195ssy_2eyo590 f19n0e5 f1ewtqcl f1jazu75 f1xeqee6 f1dxfoyt f2krc9w f1wfn5kd f1g4hkjv f15ngxrw fjbbrdp f1t94bn6 feu1g3u f1uorfem fw60kww f4xjyn1 ff1wgvm fiob0tu f1j6scgf f1x4h75k f1facbz3 f1u0rzck"
          >
            <td
              class="fui-TableCell ___mwjzo80_69n7eg0 f10pi13n f3gpkru f1dxfoyt f2krc9w f15pt5es fmrv4ls f1ft4266"
            >
              build
            </td>
            <td
              class="fui-TableCell ___mwjzo80_69n7eg0 f10pi13n f3gpkru f1dxfoyt f2krc9w f15pt5es fmrv4ls f1ft4266"
            >
              compile, test
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
